sklearn/linear_model/ridge.py

```diff
@@ -957,6 +957,8 @@ def fit(self, X, y, sample_weight=None):
         """
         X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float64,
                          multi_output=True, y_numeric=True)
+        if sample_weight is not None and not isinstance(sample_weight, float):
```

Comment:
This was based on the docstring (line 951) which said that `sample_weight` could accept a float